### DISCUSSION TITLE
Transparent background for apps in iframe

### DISCUSSION THREAD

**Discussion by Darojax** [Budibase/budibase#12823] [Help]
Hello there!

Thanks for Budibase, it's a great tool that we use for day to day management tracking members of a 100 man community.

Our main environment of usage is as iframes in a discourse forum. One issue that I am currently trying to solve is in regards to switching between light/dark theme. In order to make this look good we would really like to implement a **transparent background for our Budibase apps**, but so far I have not been able to create this. Is it even possible?

thankful for assistance,

Daniel F
Community Manager

**Comment by ConorWebb96** [comment]
Hey @Darojax,

Either for in the builder or your screens within the design area, there are a select set of themes which Budibase uses.

**Builder**
<img width="436" alt="Screenshot 2024-01-22 at 08 33 13" src="https://github.com/Budibase/budibase/assets/126772285/a25c4aad-b22f-4dda-941c-5df336ec1756">

**Design (Screens)**
<img width="316" alt="Screenshot 2024-01-22 at 08 35 43" src="https://github.com/Budibase/budibase/assets/126772285/5c1c02f7-4007-48bd-a768-ffa90088941e">

Normally I would suggest using an embed component to forcefully set the background colour. However, this will impact how easily components are seen. So I will demonstrate this as long as you are aware that the components colours etc change depending on the theme used and there isn't really a way to force these to be inline when you are overwriting the styling like this.

Put the below inside an `embed component` at the top of your screen. This should overwrite the background styling to become transparent. Haven't fully tested this with an iframe. So there may be additional classes/ids you need to force to transparent. But this seems to work.
```html
<style>
.screen-wrapper.layout-body, .nav-wrapper, #spectrum-root {
background-color: transparent!important;
}
</style>
```
**Output**
<img width="1508" alt="Screenshot 2024-01-22 at 08 51 11" src="https://github.com/Budibase/budibase/assets/126772285/272e2b76-a432-445a-a758-29262b2c6c0d">

Hopefully this helps.

**Reply by Darojax** [reply]
I tried your suggestion, but I'm slowly coming to grips with iframes not being sensitive at all to the parent platform appearance, so however one does it setting it to transparent will only be in relation to the iframe content itself, but not to it's parent medium. Other solutions would be syncing with javascipt, or even using API calls, but I'll hold off on that for now. : ) In any case thankyou for the time you took to try and assist!
